src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts

```diff
@@ -211,10 +211,47 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 			this._changeRef.detectChanges();
 		}));
 		this._register(this.cellModel.onCellPreviewModeChanged(preview => {
+			// On preview mode change, get the cursor position (get the position only when the selection node is a text node)
+			if (window.getSelection() && window.getSelection().focusNode?.nodeName === '#text' && window.getSelection().getRangeAt(0)) {
+				let selection = window.getSelection().getRangeAt(0);
+				if (selection.startOffset !== this.cellModel.richTextCursorPosition?.startOffset) {
```

Comment:
Why is this check necessary? Is this an expected scenario for the startOffset to be different in this case? Please add comments explaining this if this is accurate